ssc/cmod_wind_landbosse.cpp

```diff
@@ -0,0 +1,350 @@
+    }
+}
+
+const size_t BUFSIZE = 4096;
+
+std::string cm_wind_landbosse::call_python_module(const std::string& input_dict_as_text){
+    std::promise<std::string> python_result;
+    std::future<std::string> f_completes = python_result.get_future();
+    std::thread([&]
+                {
+                    std::string cmd = std::string(get_python_path()) + "/" + python_exec_path + " -c \"" + python_run_cmd + "\"";
+                    size_t pos = cmd.find("<input>");
+                    cmd.replace(pos, 7, input_dict_as_text);
+
+                    FILE *file_pipe = popen(cmd.c_str(), "r");
+                    if (!file_pipe){
+                        python_result.set_value("wind_landbosse error. Could not call python with cmd:\n" + cmd);
+                        return;
+                    }
+
+                    std::string mod_response;
+                    char buffer[BUFSIZE];
+                    while (fgets(buffer, sizeof(buffer), file_pipe)){
+                        mod_response += buffer;
+                    }
+                    pclose(file_pipe);
+                    if (mod_response.empty())
+                        python_result.set_value("LandBOSSE error. Function did not return a response.");
+                    else
+                        python_result.set_value(mod_response);
+                }
+    ).detach();
+
+    std::chrono::system_clock::time_point time_passed
+            = std::chrono::system_clock::now() + std::chrono::seconds(60 * 5);
+
+    if(std::future_status::ready == f_completes.wait_until(time_passed))
+        return f_completes.get();
+    else
+        throw exec_error("wind_landbosse", "python handler error. Python process timed out.");
+}
+
+#ifdef __WINDOWS__
+std::string cm_wind_landbosse::call_python_module_windows(const std::string& input_dict_as_text) {
+	STARTUPINFO si;
+	SECURITY_ATTRIBUTES sa;
+	PROCESS_INFORMATION pi;
+    HANDLE g_hChildStd_IN_Rd = NULL;
+    HANDLE g_hChildStd_OUT_Wr = NULL;
+    HANDLE g_hChildStd_OUT_Rd = NULL;
+    HANDLE g_hChildStd_IN_Wr = NULL;
+    HANDLE stderr_rd = NULL;
+    HANDLE stderr_wr = NULL;  //pipe handles
+	char buf[BUFSIZE];           //i/o buffer
+	memset(buf, 0, sizeof(buf));
+
+	std::string pythonpath = std::string(get_python_path()) + "\\" + python_exec_path;
+	CA2T programpath( pythonpath.c_str());
+	std::string pythonarg = " -c \"" + python_run_cmd + "\"";
+	size_t pos = pythonarg.find("<input>");
+	pythonarg.replace(pos, 7, input_dict_as_text);
+	CA2T programargs(pythonarg.c_str());
+
+	sa.nLength = sizeof(SECURITY_ATTRIBUTES);
+	sa.bInheritHandle = TRUE;
+	sa.lpSecurityDescriptor = NULL;
+
+	std::string output;
+    if (!CreatePipe(&g_hChildStd_IN_Rd, &g_hChildStd_IN_Wr, &sa, 0)) {
+        goto error;
+    }
+    if (!CreatePipe(&g_hChildStd_OUT_Rd, &g_hChildStd_OUT_Wr, &sa, 0)) {
+        goto error;
+    }
+    if (!CreatePipe(&stderr_rd, &stderr_wr, &sa, 0)) {
+        goto error;
+    }
+            
```

Comment:
Per the Microsoft example we should call SetHandleInformation to set g_hChildStd_IN_Wr, g_hChildStd_OUT_Rd, and stderr_rd to ensure that they are not inherited.

https://docs.microsoft.com/en-us/windows/win32/procthread/creating-a-child-process-with-redirected-input-and-output

Reply:
added